assets/src/common/components/with-enforced-file-type.js

```diff
@@ -0,0 +1,90 @@
+/**
+ * External dependencies
+ */
+import { isEqual } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import { EnforcedFileTypeToolbarSelect, getSelectMediaFrame } from './select-media-frame';
+
+const { wp } = window;
+
+/**
+ * Gets a wrapped version of MediaUpload, to enforce that it has the correct file type.
+ *
+ * Only intended for the MediaUpload in the Core Video block and the AMP Story 'Background Media' control.
+ * Though this will also apply to any other MediaUpload with allowedTypes of [ 'video' ].
+ * Partly copied from customize-controls.js.
+ *
+ * @param {Function} InitialMediaUpload The MediaUpload component, passed from the filter.
+ * @return {Function} The wrapped component.
+ */
+export default ( InitialMediaUpload ) => {
+	/**
+	 * Partly copied from customize-controls.js.
+	 *
+	 * @see wp.media.HeaderControl
+	 */
+	return class EnforcedFileTypeMediaUpload extends InitialMediaUpload {
+		/**
+		 * Constructs the class.
+		 */
+		constructor() {
+			super( ...arguments );
+
+			// This class should only be present in the MediaUpload for the AMP Story 'Background Media' or when only 'video' types are allowed, like in the Core Video block.
+			if ( 'story-background-media' === this.props.id || isEqual( [ 'video' ], this.props.allowedTypes ) ) {
```

Comment:
`isEqual( [ 'video' ], this.props.allowedTypes` is the only way I could find to select the Core 'Video' block. Though this will of course also apply to any other `MediaUpload` where that happens to be true.

Reply:
That seems fine because MP4 is the only format supported in stories.